### PR TITLE
Fix parsing of declaration in non declaration contexts

### PR DESCRIPTION
**Pull request by arv**
Declarations (let, const, function, class) are not allowed as
Statements. For example, the following is not valid:

  if (true)
    let x = 1;

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

**Comment by arv** [comment]
This TODO finally got in my way.

File: src/syntax/Parser.js [hunk 1]
```diff
@@ -1212,9 +1220,6 @@ export class Parser {
    * @private
    */
   parseSpreadExpression_() {
-    if (!this.options_.spread)
```
Comment (johnjbarton) [inline src/syntax/Parser.js]: This does not seem related, was it intended?

**Comment by johnjbarton** [comment]
LGTM

File: src/syntax/Parser.js (hunk 1 above)
Comment (arv) [inline src/syntax/Parser.js]: This is a cleanup. Both callers do a peekSpread_ first.
